### PR TITLE
Alert a11y 118n fixes

### PR DESCRIPTION
**Pull request by andreapiras**
## 🖼 Context

Fixing up a11y / i18n issues for the Alert component. 

## 🚀 Changes

* Added a new translations prop to pass down aria labels / other translations. See [this RFC](https://github.com/transferwise/neptune-web/pull/527) for more.
* We also realised while fixing eslint errors that we needed to add React Testing Library as a peer dep, because you can't rely on the dependencies of a dependency. Also updated `test-config` and `dynamic-flows` to fix this.

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

**Comment by sbrandwoo** [comment]
```
 <Alert translations={{ ariaLabel: 'Something' }} />
```
From just reading this definition, it looks like you are labelling the Alert, not labelling the close button inside.

File: packages/components/src/alert/Alert.js [hunk 1]
```diff
@@ -1,73 +1,79 @@
...
+        alertConfigClasses,
+        arrowClasses(arrow),
+        className,
+      )}
+    >
+      {dismissible && (
+        <button
+          type="button"
+          className="close"
+          data-dismiss="alert"
```
[the remark this answers is not in the document]
Comment (andreapiras) [reply packages/components/src/alert/Alert.js:66]: @dozortsev

[the remark this answers is not in the document]
Comment (dozortsev) [reply packages/components/src/alert/Alert.js:54]: we copy pasted this from Bootstrap's implementation https://bootstrap.transferwise.com/components/#alerts-dismissible

not sure why it's necessary there

[the remark this answers is not in the document]
Comment (andreapiras) [reply packages/components/src/alert/Alert.js:54]: I think it was related to the old bootstrap js
image

[the remark this answers is not in the document]
Comment (dozortsev) [reply packages/components/src/alert/Alert.js:54]: yeah, seems can be delete here

**Review by helendurranttw — APPROVED** [review]

**Pull request merged**